modules/cluster-compare-templating-reference.adoc

```diff
@@ -0,0 +1,115 @@
+// Module included in the following assemblies:
+
+// *scalability_and_performance/cluster-compare/creating-a-reference-configuration.adoc
+
+:_mod-docs-content-type: REFERENCE
+
+[id="cluster-compare-templating-reference_{context}"]
+= Reference template functions
+
+The `cluster-compare` plugin supports all `sprig` library functions, except for the `env` and `expandenv` functions. For the full list of `sprig` library functions, see "Sprig Function Documentation".
+
+The following table describes the additional template functions for the `cluster-compare` plugin:
+
+.Additional cluster-compare template functions
+[cols=".^2,.^4,.^6a",options="header"]
+|====
+
+|Function |Description |Example
+
+|`fromJson`
+|Parses the incoming string as a structured JSON object.
+|`value: {{ obj := spec.jsontext \| fromJson }}{{ obj.field }}`
+
+|`fromJsonArray`
+|Parses the incoming string as a structured JSON array.
+| `value: {{ obj := spec.jsontext \| fromJson}}{{ index $obj 0 }}`
+
+|`fromYaml`
+|Parses the incoming string as a structured YAML object.
+|`value: {{ obj := spec.yamltext \| fromYaml }}{{ obj.field }}`
+
+|`fromYamlArray`
+|Parses the incoming string as a structured YAML array.
+|`value: {{ obj := spec.yamltext \| fromYaml}}{{ index $obj 0 }`
+
+|`toJson`
+|Renders incoming data as JSON while preserving object types.
+|`jsonstring: {{ $variable \| toJson }}`
+
+|`toToml`
+|Renders the incoming string as structured TOML data.
+|`tomlstring: {{ $variable \| toToml }}`
+
+|`toYaml`
+|Renders incoming data as YAML while preserving object types.
+|For simple scalar values: `value: {{ $data \| toYaml }}`
+
+For lists or dictionaries: `value: {{ $dict \| toYaml \| nindent 2 }}`
+
+|`doNotMatch`
+|Prevents a template from matching a cluster resource, even if it would normally match. You can use this function inside a template to conditionally exclude certain resources from correlation. The specified reason is logged when running with the `--verbose` flag. Templates excluded due to `doNotMatch` are not considered comparison failures.
+
+This function is especially useful when your template does not specify a fixed name or namespace. In such cases, you can use the `doNotMatch` function to exclude specific resources based on other fields, such as labels or annotations.
```

Comment:
```suggestion
This function is especially useful when your template does not specify a fixed name or namespace. In these cases, you can use the `doNotMatch` function to exclude specific resources based on other fields, such as `labels` or `annotations`.
```
Are labels and annotations fields? Sounds like it, so wonder if these are params that should be backticked?